tests/test_modeling_common.py

```diff
@@ -4829,6 +4831,73 @@ def test_forward_with_num_logits_to_keep(self):
             # Assert the last tokens are actually the same (except for the natural fluctuation due to order of FP ops)
             self.assertTrue(torch.allclose(all_logits[:, -1:, :], last_token_logits, atol=1e-5))
 
+    @slow
+    @require_torch_greater_or_equal("2.3")
+    def test_torch_export(self, config_kwargs=None):
+        if not self.test_torch_exportable:
+            self.skipTest(reason="Torch export test is not enabled for this model.")
+
+        def recursively_check(eager_outputs, exported_outputs):
+            is_tested = False
+            if isinstance(eager_outputs, torch.Tensor):
+                self.assertEqual(eager_outputs.shape, exported_outputs.shape)
+                self.assertTrue(torch.allclose(eager_outputs, exported_outputs, atol=1e-4))
+                return True
+            elif isinstance(eager_outputs, (tuple, list)):
+                for eager_output, exported_output in zip(eager_outputs, exported_outputs):
+                    is_tested = is_tested or recursively_check(eager_output, exported_output)
+                return is_tested
+            return is_tested
+
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        inputs_dict["return_dict"] = False
+
+        # override config kwargs for torch export if needed
+        config_kwargs = config_kwargs if config_kwargs is not None else {}
+        for key, value in config_kwargs.items():
+            setattr(config, key, value)
+
+        for model_class in self.all_model_classes:
+            if model_class.__name__.endswith("ForPreTraining"):
+                continue
+
+            with self.subTest(model_class.__name__):
+                model = model_class(config).eval().to(torch_device)
+
+                # Prepare exported program inputs based on model.forward function signature
+                signature = inspect.signature(model.forward)
+                kwarg_names = [
+                    name
+                    for name, param in signature.parameters.items()
+                    if param.default is not inspect.Parameter.empty
+                ]
+                arg_names = [name for name in signature.parameters.keys() if name not in kwarg_names]
+                inputs_args = tuple(inputs_dict[name] for name in arg_names if name in inputs_dict)
```

Comment:
I think it's better to check all name in  `arg_names` are in `inputs_args` otherwise fail it early here.

Reply:
There might be situations when not all `arg_names` are in `inputs_dict`, e.g., the model might accept `pixel_mask` but we can avoid providing it, in such cases, it will be created in the modeling code as `torch.ones`. (if I got your comment correctly)

Reply:
Those in `arg_names` doesn't have any default value (not even `None`), so could we call that method without passing such arguments? (but let me know if I think it wrong)

Reply:
The idea here is to separate `inputs_dict` to the positional and keyword arguments based on `forward` function signature + provide them in the correct order. If any name not exists in `inputs_dict` it's fine, inputs_idct defines all required arguments to run model's forward.

Reply:
OK, I think my brain is a bit too linear and thinking

> separate inputs_dict to ....
is 
> inputs_args = tuple(inputs_dict[name] for name in inputs_dict if name in arg_names)

but they are the same .... sorry I am a literal person.